esmf-semantic-aspect-meta-model/src/main/resources/samm/characteristic/2.0.0/characteristic-shapes.ttl

```diff
@@ -783,3 +783,26 @@ samm-c:FixedPointConstraintShape
       sh:name "integer" ;
       sh:description "The number of integral digits for a fixed point number." ;
    ] .
+
+samm-c:TimeSeriesShape
+   a sh:NodeShape ;
+   sh:targetClass samm-c:TimeSeries ;
+   sh:property [
+      sh:path samm:dataType ;
+      sh:minCount 0 ;
+      sh:maxCount 1 ;
+      sh:name "dataType" ;
+      sh:description "The current data tуpe is the child of TimeSeries or its derived types." ;
+      sh:sparql [
+         a sh:SPARQLConstraint ;
+         sh:message "Entry '{$this}' to be extended from 'TimeSeriesEntity'. Current type '{?definedDataType}' doesn't not extended from 'TimeSeriesEntity'. You can use this structure for resolve it: 'samm:extends samm-e:TimeSeriesEntity ;'." ;
+         sh:prefixes samm:prefixDeclarations ;
+         sh:select """
+            select $this ?definedDataType
+            where {
+               $this samm:dataType ?definedDataType .
+               filter ( not exists { ?definedDataType samm:extends/samm:extends* samm-e:TimeSeriesEntity } ) .
```

Comment:
The expression `samm:extends/samm:extends*` can be simplified to `samm:extends+`, see [SPARQL path expressions](https://www.w3.org/TR/sparql11-query/#propertypaths).

Reply:
Thanks, it was resolved!